api/src/pcapi/core/users/api.py

```diff
@@ -1356,7 +1356,14 @@ def _get_users_with_suspended_account_to_notify(expiration_delta_in_days: int) -
     ).with_entities(users_models.User)
 
 
-    return query
+def get_suspended_upon_user_request_accounts_since(expiration_delta_in_days: int) -> Query:
+    start = datetime.date.today() - datetime.timedelta(days=expiration_delta_in_days)
+    user_ids_and_latest_action = _get_users_with_suspended_account()
+
+    return user_ids_and_latest_action.filter(
+        history_models.ActionHistory.actionDate <= start,
+        history_models.ActionHistory.extraData["reason"].astext == constants.SuspensionReason.UPON_USER_REQUEST.value,
+    ).with_entities(users_models.User)
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 issue:** ‏Là tu déplace la fonction en faisant une refacto en même temps, c'est dur à lire

L'idéal: 

- un commit qui déplace 
- un commit qui refactore